### PR TITLE
US-6 Sensors & US-7 Celestial Map

### LINKED ISSUE #7: US-6 Sensors
Estimate: 5 hrs
Actual:

User Story: US-6
Sensors
Priority: 1

Story: As a player, I want to see what is located at nearby Celestial Points, so I know where things are.

Acceptance Criteria:
1. Verify that if the player deploys sensors for the current CP, 2% of the supplies are consumed.
2. Verify that celestial objects within two CP of the current CP are displayed
3. Verify that celestial objects within two CP of the current CP are added to the Celestial Map

### LINKED ISSUE #8: US-7 Celestial Map
Estimate:
Actual:

User Story: US-7
Celestial Map
Priority: 1

Story: As a player, I want to see all the cells in the game map I have visited and what is located in each cell upon request, so I can plan my next moves

Acceptance Criteria:
1. Verify that before any sensor activity is initiated, the Celestial Map shows Celeron, Xeon, and Ryzen
2. Verify that each time sensor activity is initiated the artifacts shown are added to the celestial map

### PR DESCRIPTION
**Pull request by halyngoc**
Closes #7, closes #8

New to-do:
- [x] Resolve conflicts
- [x] More testing
- [x] ~Deal with #21~ _moving this to #22_

Normally I'd hate making one PR do 2 things but they're dependent on each other so idk how else I can do this.

Also while working on these USs I found 3 words that seemingly mean the same thing: _celestial bodies_, _celestial objects_, and _celestial artifacts_. I'll assume they're the same until someone tells me otherwise. I pick celestial object (CO).

The 3 planets (Celeron, Xeon, and Ryzen) have been added to the Celestial Map (CM) as simple strings, but as far as I know other behaviors from these planets are not implemented in these USs.

❓ At the start of the game, the CM shows the CPs where the 3 planets are. If there are any other COs in those CPs, the CM also shows them. Idk if this is intended behavior but I don't see anything in the acceptance criteria that contradicts it.

To-do for sensors:
- [x] Display the current CP (Celestial Point) at the start of the game and on every move
- [x] Display all nearby CPs when the sensors are deployed (_turns_ are not part of these USs, but supplies are consumed as if a turn has passed)
- [x] Consume supplies when sensors are deployed
- [x] Add COs found to CM
- [x] Verify (see #7 for acceptance criteria)

To-do for CM:
- [x] Implement CM
- [x] Add Celeron, Xeon, and Ryzen on the CM
- [x] Verify (see #8 for acceptance criteria)

### CODE REVIEW

File: gameObjects.js [hunk 1]
```diff
@@ -75,14 +80,29 @@ function setData() {
...
 //Map could contain 128x128 celestialPoint() objects
-var gameSpace = {
-  map : [spaceship.maxCoord + 1][spaceship.maxCoord + 1],
-}
+var gameSpace = [];
+window.onload = function() {
+  // Initialize gameSpace
+  for (let i = 0; i <= spaceship.maxCoord; i++) {
+    var CPRow = [];
+    for (let i = 0; i <= spaceship.maxCoord; i++) {
```
[the remark this answers is not in the document]
Comment (halyngoc) [reply gameObjects.js]: Shut up hound bot `let` is better than `var` 😃

**Review by SimonBartonPSU — COMMENTED** [review]
I cloned this branch and did some testing in my Chrome browser. This code passes all of the acceptance criteria for US-6 and US-7. Well done.

**Comment by halyngoc** [comment]
Attempted to resolve conflicts. I'll test to see if anything is broken.

Edit: Nvm everything is broken 👍

**Comment by halyngoc** [comment]
It seems that everything is working now. Except that #21 thing. I think it should be done in #22 after this PR gets merged.

**Review by SimonBartonPSU — APPROVED** [review]
Good changes all around. Clean functions for CM and sensors.

I tested this in my browser as well and previous acceptance tests are functional. Let's get this merged. Liam's new dev menu can merge next then I can fix merge conflicts for #22 in a few minutes since its a small change.

**Pull request merged**